MSAL/src/MSALAccount.m

```diff
@@ -76,36 +73,58 @@ - (id)initWithUsername:(NSString *)username
         _homeAccountId = [[MSALAccountId alloc] initWithAccountIdentifier:homeAccountId
                                                                  objectId:uid
                                                                  tenantId:utid];
-        
-        _localAccountId = [[MSALAccountId alloc] initWithAccountIdentifier:localAccountId
-                                                                  objectId:localAccountId
-                                                                  tenantId:tenantId];
 
         _lookupAccountIdentifier = [[MSIDAccountIdentifier alloc] initWithDisplayableId:username homeAccountId:homeAccountId];
+        
+        if (tenantProfiles.count > 0)
+        {
+            self.mTenantProfiles = [[NSMutableArray alloc] initWithArray:tenantProfiles];
+        }
     }
 
     return self;
 }
 
-- (id)initWithMSIDAccount:(MSIDAccount *)account
+- (instancetype)initWithMSIDAccount:(MSIDAccount *)account
+                createTenantProfile:(BOOL)createTenantProfile
 {
+    NSError *error;
+    MSALAuthority *authority = [MSALAuthorityFactory authorityFromUrl:account.authority.url context:nil error:&error];
+    if (error || !authority)
+    {
+        MSID_LOG_ERROR(nil, @"Failed to create msal authority from msid authority!");
+        return nil;
+    }
+    
+    NSArray *tenantProfiles;
+    if (createTenantProfile)
+    {
+        MSALTenantProfile *tenantProfile = [[MSALTenantProfile alloc] initWithUserObjectId:account.localAccountId
```

Comment:
nit: check that tenant profile creation succeeded

Reply:
:+1: changed